rayon-core/src/thread_pool/mod.rs

```diff
@@ -50,6 +50,35 @@ impl ThreadPool {
         }
     }
 
+    /// Executes `op` within the threadpool without blocking the
+    /// calling thread. Any attempts to use `join`, `scope`, or
+    /// parallel iterators will then operate within that
+    /// threadpool.
+    ///
+    /// # Warning: thread-local data
+    ///
+    /// Because `op` is executing within the Rayon thread-pool,
+    /// thread-local data from the current thread will not be
+    /// accessible.
+    ///
+    /// # Panics
+    ///
+    /// Polling the [`InstallFuture`] will propagate panics from
+    /// `op`.
+    ///
+    /// [`InstallFuture`]: struct.InstallFuture.html
+    #[cfg(feature = "unstable")]
+    pub fn install_future<OP, R>(&self, op: OP) -> InstallFuture<OP, R>
+        where OP: FnOnce() -> R + Send
+    {
+        unsafe {
+            let job_a = StackJob::new(op, LockLatch::new());
+            self.registry.inject(&[job_a.as_job_ref()]);
+
+            InstallFuture { job: Some(job_a) }
```

Comment:
I don't think this is safe at all.  A `StackJob` must stay at the **exact** same memory location until completion, because you're handing its pointer to the registry.  When you move it to the `InstallFuture` and return, the registry is left with a dangling pointer!